python-avd/pyavd/_eos_designs/structured_config/base/__init__.py

```diff
@@ -331,6 +331,9 @@ def spanning_tree(self) -> None:
         if self.shared_utils.node_config.spanning_tree_mst_pvst_boundary:
             self.structured_config.spanning_tree.mst.pvst_border = True
 
+        if stp_po_range := self.shared_utils.node_config.spanning_tree_port_id_allocation_port_channel_range:
+            self.structured_config.spanning_tree.port_id_allocation_port_channel_range._update(minimum=stp_po_range.minimum, maximum=stp_po_range.maximum)
```

Comment:
I think you can just assign it directly, since the class is a pure $ref. You can see in `schema/__init__.py` that it is using the EosCliConfigGen class directly.
```suggestion
            self.structured_config.spanning_tree.port_id_allocation_port_channel_range = stp_po_range
```

Reply:
updated, type hint is also happy